File: backend/src/main/java/group8/bloodbank/model/BloodBankBlood.java
package group8.bloodbank.model;

import com.google.common.primitives.UnsignedInteger;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;

@Entity
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Table(name="blood_bank_blood")
public class BloodBankBlood {


    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column
    private Integer A_POSITIVE;

    @Column
    private Integer A_NEGATIVE;

    @Column
    private Integer B_POSITIVE;

    @Column
    private Integer B_NEGATIVE;

    @Column
    private Integer O_POSITIVE;

    @Column
    private Integer O_NEGATIVE;

    @Column
    private Integer AB_POSITIVE;

    @Column
    private Integer AB_NEGATIVE;

}

File: frontend/src/app/model/bloodBankBlood.ts

export class BloodBankBlood {
    A_POSITIVE!: number; 
    A_NEGATIVE!: number;
    B_POSITIVE!: number;
    B_NEGATIVE!: number;
    AB_POSITIVE!: number;
    AB_NEGATIVE!: number;
    O_POSITIVE!: number;
    O_NEGATIVE!: number;
  

  public constructor(obj?: any) {
      if (obj) {
          this.A_POSITIVE = obj.A_POSITIVE
          this.A_NEGATIVE = obj.A_NEGATIVE;
          this.B_POSITIVE = obj.B_POSITIVE;
          this.B_NEGATIVE = obj.B_NEGATIVE;
          this.AB_POSITIVE = obj.AB_POSITIVE;
          this.AB_NEGATIVE = obj.AB_NEGATIVE;
          this.O_POSITIVE = obj.O_POSITIVE;
          this.O_NEGATIVE = obj.O_NEGATIVE;
      }
  }
}
